backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -311,13 +306,16 @@ private List<Integer> getOnlineCores(List<VdsCpuUnit> cpuTopology) {
     private List<VdsCpuUnit> getMaxFreedSocket(List<VdsCpuUnit> cpuTopology) {
```

Comment:
The method is a bit confusing as it does two different things - finds the socket with max free cpus + returns all cpus for that socket. I think it could be changed to return just the socket id of the socket with max free cpus and in the code it could be called like:
```
getCoresInSocket(cpuTopology, getMaxFreedSocket(cpuTopology))
```